src/common/checksums.h

```diff
@@ -161,9 +167,12 @@ class OCSYNC_EXPORT ValidateChecksumHeader : public QObject
      */
     void start(std::unique_ptr<QIODevice> device, const QByteArray &checksumHeader);
 
+    QByteArray calculatedChecksumType() const;
+    QByteArray calculatedChecksum() const;
+
 signals:
     void validated(const QByteArray &checksumType, const QByteArray &checksum);
-    void validationFailed(const QString &errMsg);
+    void validationFailed(const QString &errMsg, FailureReason reason);
```

Comment:
is there a reason not to also add `calculatedChecksumType` and `calculatedChecksum` to avoid doing a `qobject_cast` down the road ?

Reply:
@mgallien Isn't it more verbose to have additional arguments?

Reply:
Might be overkill but since we are overloading functions here a struct could be used to hold all of this data? Could be reused for the `validated` method too

Reply:
my point is that while `qobject_cast` might work, it is far from ideal to rely on it to be set while we could just add two extra args

Reply:
@mgallien That implies more changes, but, ok